course5/homework09_2/main.cc

```diff
@@ -0,0 +1,116 @@
+/*
+Write a simple single-object implementation of std::unique_ptr, complete with make_unique.
+*/
+#include <iostream>
+#include <string>
+#include <cassert>
+
+
+// operators new/delete or new[]/delete[]
+
+template<typename T>
+struct MoreUniquePtr
+{
+	MoreUniquePtr() = default;
+	MoreUniquePtr(const T& src)
```

Comment:
Is this what std::unique_ptr is doing?

And, such a constructor, not marked as 'explicit', would be very dangerous, could lead to unintended code to compile.